mbson/bson_raw.go

```diff
@@ -0,0 +1,29 @@
+package mbson
+
+import (
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+// RawLookup combines bson.Raw’s LookupErr method with an additional
+// unmarshal step. The result is a convenient way to extract values from
+// bson.Raw. The returned boolean indicates whether the value was found.
+func RawLookup[T any](doc bson.Raw, dest *T, keys ...string) (bool, error) {
```

Comment:
(For the record, I still find this function name extremely confusing.)

Reply:
Noted … but the functionality seems just too useful not to bring over.